WTG.Analyzers/Analyzers/CodeContracts/CodeContractsHelper.cs

```diff
@@ -370,5 +396,39 @@ public override void VisitLocalFunctionStatement(LocalFunctionStatementSyntax no
 			readonly SemanticModel model;
 			readonly CancellationToken cancellationToken;
 		}
+
+		sealed class ContractForAllVisitor : CSharpSyntaxWalker
+		{
+			public ContractForAllVisitor(SemanticModel model, CancellationToken cancellationToken)
+			{
+				this.model = model;
+				this.cancellationToken = cancellationToken;
+			}
+
+			public bool EncounteredContractForAll { get; private set; }
+
+			public override void Visit(SyntaxNode node)
+			{
+				if (!EncounteredContractForAll)
+				{
+					base.Visit(node);
+				}
+			}
+
+			public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+			{
+				var symbol = model.GetSymbolInfo(node, cancellationToken).Symbol;
+				if (symbol != null && symbol.Name == "ForAll" && symbol.ContainingType.IsMatch("System.Diagnostics.Contracts.Contract"))
+				{
+					EncounteredContractForAll = true;
+					return;
```

Comment:
Do you think we need to worry about nested calls to ForAll()?

Reply:
I doubt it...